### DISCUSSION TITLE
How to keep toast alive?

### DISCUSSION THREAD

**Discussion by nneto** [cdmoro/bootstrap-vue-3#747] [Q&A] [answered]
I tried:

```
<b-container
  :toast="{ root: true }"
  position="position-fixed"
  auto-hide="false"
></b-container>
```

Also

```
toast.show(
  {
    title: "Lorem ipsum",
    body: "Lorem ipsum dolor sit amet",
    autoHide: false,
  },
  {
    autoHide: false,
    variant: "info",
  }
)
```

But toast keep hiding after 5s. Vue.js Dev Tools plugin for the browser also shows the component with prop autoHide set to true.

**Comment by VividLemon** [comment, marked as answer]
Toasts are broken and are being fixed in my unreleased changes https://github.com/cdmoro/bootstrap-vue-3/compare/main...VividLemon:bootstrap-vue-3:main . I am thinking about removing the component outright and replacing it with a https://quasar.dev/quasar-plugins/notify like plugin

The point is I'm working on it right now

**Reply by nneto** [reply]
Got it! Thank you!

**Reply by nneto** [reply]
The toasts being broken is also the reason of why I'm having trouble trying to show toasts from composables?
I get an error when `const toast = useToast()` in composables. The error occurs from:
```
2192 function qa(e, t = _t) {
2193   const a = Oe(so());
2194   if (!e)
2195     return new Zt(a.getOrCreateViewModel());
2196   const l = { id: Symbol("toastInstance") }, n = { ...oo, ...l, ...e }, i = a.getOrCreateViewModel(n);
2197   return new Zt(i);
2198 }
```
This is what happens:
![image](https://user-images.githubusercontent.com/7104467/196274275-42fe41ee-3061-4ee6-8263-4da339513f44.png)

**Reply by VividLemon** [reply]
The composable is not the same as the component. They are both broken, for separate reasons.

**Comment by aceofwings** [comment]
I can assist if you have questions @VividLemon. I don't think it is wise pulling from another framework and adding it as a dependency to get notification capabilities. I recommend dumbing down what I initially wrote(I was trying to for multiple container support. Seeing that toasts only needs to render on the 1 container for most of the use cases all the logic can be dumbed down to a simple array attached to a designated b-container

**Reply by VividLemon** [reply]
> I don't think it is wise pulling from another framework and adding it as a dependency to get notification capabilities

No, the goal is to make a Bootstrap toast that is _modelled_ after the quasar notification. To have it be used in the same way. By using a simple method to show, hide, adjust options, etc. I think they pulled off their system quite well, and it works well. So, the usability would be similar, but built from scratch and is Bootstrap's. 

So, I was trying to get the toast component to work the other day, but was having some issues with it. The main issue is that the parent container (toaster) does not exist when the toast is mounted. Boostrap-vue handled this by on the fly creating a container. However, they had the assistance of using portal-vue Wormhole, to check if one existed. Since Vue 3 doesn't have that (that I know of), <teleport> will fail. The problem with using b-container as a mounting point for this is the non-existence of a b-container component. Vuetify accomplishes this by requiring v-app, we don't have the luxury of forcing someone to use b-container... Though a "b-app" could be "possible", but not preferred. 

Despite all of this, the actual component seems unnecessary and I think it's better to scrap it, create a plugin from scratch, make it super useful, and a worthwhile replacement for anyone using the component. The only problem of using the plugin is that the functionality for a _static_ toast , ie one rendered in place, may be lost. However, I think I can workaround this by essentially allowing the <teleport> inside the plugin to accept user fed values that aren't explicit in the position argument. (note, teleports cause problems if the "to" doesn't exist **before** a teleport occurs. And it will not auto update itself if the target exists at a later time. It is best to either create the target, or target a true html target, usually the body tag)

Ps keep an eye out for some emails or others about the possible upcoming merging with bootstrap-vue. Lmk if you want to be included in the discord dm.

**Reply by aceofwings** [reply]
ok, appending to document as it was original and how other solutions do it makes sense.

**Reply by aceofwings** [reply]
@VividLemon you can get me In the discord that would be great

**Reply by VividLemon** [reply]
> @VividLemon you can get me In the discord that would be great

Add me on https://discord.com/ so we can talk! My username is TempleOS#1085.
